src/decoder.rs

```diff
@@ -1356,16 +1384,16 @@ mod tests {
 
     fn accepts(d: &Decoder, input: &[u8], tail: &[u8], expect: Value) {
         let program = Program::new();
-        let mut scope = Scope::new();
-        let (val, remain) = d.parse(&program, &mut scope, ReadCtxt::new(input)).unwrap();
+        let scope = Scope::Empty;
+        let (val, remain) = d.parse(&program, &scope, ReadCtxt::new(input)).unwrap();
         assert_eq!(val, expect);
         assert_eq!(remain.remaining(), tail);
     }
 
     fn rejects(d: &Decoder, input: &[u8]) {
         let program = Program::new();
-        let mut scope = Scope::new();
-        assert!(d.parse(&program, &mut scope, ReadCtxt::new(input)).is_err());
+        let scope = Scope::Empty;
```

Comment:
Likewise, we don't necessarily need a name for `Scope::Empty` if it is immutable and only used in one place. Unless I am wrong about lifetime inference for temporary values, that is.

Reply:
Fixed.